1-Draft/RFCXXXX-Send-MailMessage.md

```diff
@@ -0,0 +1,72 @@
+---
+RFC: RFCXXXX
+Author: Travis Plunk
+Status: Draft
+SupercededBy: N/A
+Version: 1.0
+Area: cmdlets
+Comments Due: 4/30/2019
+Plan to implement: No
+---
+
+# Deprecating `Send-MailMessage`
+
+`Send-MailMessage` does not support many modern protocols leading to the inability to use this with modern secure mail services.
+See [DotNet DE0005](https://github.com/dotnet/platform-compat/blob/master/docs/DE0005.md).
+Since the protocols it uses are no longer considered secure, the cmdlet as is should not be used.
+I recommend that we implement a plan to remove the cmdlet from powershell core and, if needed,
+re-implement as a separate module.
+
+## Motivation
+
+    Most major mail platforms now have REST methods to send mail allowing existing cmdlets to allow sending mail.
+
+    * gmail
+      * https://developers.google.com/gmail/api/v1/reference/users/messages/send
+      * https://stackoverflow.com/questions/24460422/how-to-send-a-message-successfully-using-the-new-gmail-rest-api
+    * Office365
+      * https://docs.microsoft.com/en-us/previous-versions/office/office-365-api/api/version-2.0/mail-rest-operations#SendMessages
+
+## Removal Plan
+
+### Add a warning to 6.2 after the release of 6.2
+
+In a 6.2 servicing release, we will add a warning to `Send-MailMessage` that an alternative method should be found.
+
+### Remove the cmdlet in 6.3
+
+During the development of 6.3, remove the `Send-MailMessage` cmdlet.
```

Comment:
There were 4 PRs to improve Send-MailMessage over the past half year. That indicates people are actively using this cmdlet.
Removing this cmdlet without an available replacement seems not right ...

Reply:
I don't think if we don't remove the cmdlet, we need to add a switch... Something like `-AllowInsecure` and document that newer method should be used.

Reply:
If we have plans to update it completely, why not simply move it to its own module that comes "in the box" with PS Core, starting with the version that implements the newer recommended libraries?

Reply:
This option is mentioned in the `Alternate Proposals and Considerations` section.

Reply:
Agree with @vexx32 on moving it out to a separate module, though this should be still bundled with future PowerShell Core versions just Like PowerShellGet was extracted from the codebase and brought in as a build package, back in the early days of PowerShell going open source.

Reply:
We could disable the cmdlet by default but add option in powershell.config.json to enable it in environment where it is needed.

Reply:
First of all, I really want to be as constructive as I can. I may (and probably will) be wrong in some areas or not hit the correct dev terminology. Bear with me please. I stumbled across this thread after serious confusion was caused by the new "obsolete" message in production. I believe both the message and the plan to remove/change Send-MailMessage is wrong, and am trying to make a point below. 
- DEFINITION OF PROBLEM: I was puzzled to learn that this decision seems to be based largely on assumptions rather than facts. Is there a specific bug or security flaw in the cmdlet? Apparently not. Will the cmdlet continue to do what it promises to do? Yes it will. Is there a better replacement cmdlet available? No, there isn't. 
Fact seems to be that the underlying API does not support some new security protocols. The cmdlet is not falling back automatically or transparently to clear text auth, though, and the latter may be just fine for non-security relevant data mailings from servers which is a predominant use case in PowerShell scripts. So there is apparently no security reason to recommend stop using the cmdlet.
Another fact seems to be the dotnet team marked the underlying API obsolete. However there are apparently no plans to remove them anytime soon, and the cmdlet will continue to work for the foreseeable future. So there is no rush. There would have been plenty time to bring this discussion to an end, and then come up with an answer. Instead, a message was emitted even though it is not yet clear whether the cmdlet will be serviced, replaced, or removed. What is the intended benefit of this message?
- COMMON TERMINOLOGY: When you declare a cmdlet "obsolete" and emit a warning to the end user, it is not important what "obsolete" means to you. It is important what the addressed audience understands. In the outside world, "obsolete" means the use of this cmdlet is no longer recommended and considered unsafe practice. Whenever the authors of a software declare part of their software "obsolete", the outside world expects time to adjust before this happens (by giving a forewarning) and a transition path. Neither is the case here.
- TARGET AUDIENCE: A considerable percentage of the PowerShell user base are admins w/o c# dev background. When the suggested transition path is to use "MailKit" and involves github resources, you are essentially bubbling up unfiltered dotnet recommendations to your PowerShell audience which is not appropriate for this audience. Any remedy should work in the context of your audience, not your own. You may want to use MailKit in a new version of Send-MailMessage, but a production admin certainly may not know how to do this or may not have the permission to use the resources in the first place.
- NO BREAKING CHANGES CONTRACT: The outside expectation was that PS Core introduced fundamental breaking changes ONCE in order to address cross platform challenges. Ongoing, it was considered the team would continue to avoid breaking changes to engine AND default cmdlets  - and why shouldn't you. It was somewhat illuminating just how relatively lightheartedly you talk about removing a cmdlet that is in use in zillions of production scripts and technically can definitely be improved without breaking changes. Dropping default PowerShell cmdlets that you shipped with the engine is in some respect the same as pulling an open source repo/dependency that is in use by someone else. If you don't want that, you probably need to strip all cmdlets from the engine that aren't essential to the engine, and ship them in different modules.
However, I would advise not to. One of the strengths of PowerShell was always that it shipped with a critical number of cmdlets and enabled many automation tasks OOB. Loading additional modules/packs on demand may be common practice for devs but is prohibitive in many production environments and not feasible for many offline sites. If you strip all cmdlets from the engine, this will seriously affect the one-stop-philosophy. 
In a nutshell: I'd hope the user side will be integrated better in the decision making process which of course is in part the responsibility of the user side (i.e. people like me). PS users tend to not stumble across github discussions by nature. If you find these thoughts helpful, I'll gladly contribute. If they are not appropriate or this is not the right place, let me know and I stop.

Reply:
@TobiasPSP, I agree with all your comments.
It should also be pointed out that SMTP has not, and should not be considered a secure protocol.  While it now has lots of security enhancements, they aren't enabled by default and there should never have been an assumption of any kind of security when using SmtpClient to send mail.